File: 1436-destination-city/1436-destination-city.py
class Solution:
    def destCity(self, paths):
        visit = set()
        connection = {}

        if len(paths) == 1:
            return paths[0][1]
        location = paths[0][0]
        for item in paths:
            connection[item[0]] = item[1]

        while location not in visit and location in connection:
            visit.add(location)
            location = connection[location]

        return location

File: 1436-destination-city/1436-destination-city.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        set<string> visit;
        unordered_map<string,string> connection;
        string location = paths[0][0];
        
        if(paths.size() == 1) return paths[0][1];
        for(auto &item: paths)
            connection.insert({item[0],item[1]});
        while(visit.count(location)<1 && connection.count(location)>0){
            visit.insert(location);
            location = connection[location];
        }
        
        return location;
    }

};